language/sygus.tex

```diff
@@ -1068,29 +1083,62 @@ \subsection{Asserting Synthesis Constraints and Assumptions}
-   \paren{\vardeclkwd\mbox{ }v'_1\mbox{ }\sigma_1}\\
-   \ldots\\
-   \paren{\vardeclkwd\mbox{ }v_n\mbox{ }\sigma_n}\\
-   \paren{\vardeclkwd\mbox{ }v'_n\mbox{ }\sigma_n}\\
-   \paren{\constraintkwd\mbox{ }\paren{\texttt{=>}\mbox{ }
-   \paren{S_{pre}\mbox{ }v_1\mbox{ }\ldots\mbox{ }v_n}\mbox{ }
-   \paren{S\mbox{ }v_1\mbox{ }\ldots\mbox{ }v_n}}}\\
-   \paren{\constraintkwd\mbox{ }\paren{\texttt{=>}\mbox{ }
-   \paren{\texttt{and}\mbox{ }\paren{S\mbox{ }v_1\mbox{ }\ldots\mbox{ }v_n}\mbox{ }
-   \paren{S_{trans}\mbox{ }v_1\mbox{ }\ldots\mbox{ }v_n\mbox{ }v'_1\mbox{ }\ldots\mbox{ }v'_n}}\mbox{ }
-   \paren{S\mbox{ }v'_1\mbox{ }\ldots\mbox{ }v'_n}}}\\
-   \paren{\constraintkwd\mbox{ }\paren{\texttt{=>}\mbox{ }
-   \paren{S\mbox{ }v_1\mbox{ }\ldots\mbox{ }v_n}\mbox{ }
-   \paren{S_{post}\mbox{ }v_1\mbox{ }\ldots\mbox{ }v_n}}}\\
-\end{array}
-\]
+\begin{lstlisting}[language=SyGuS, numbers=none, basicstyle=\ttfamily\small]
+($\vardeclkwd$ $v_1$ $\sigma_1$)
+($\vardeclkwd$ $v'_1$ $\sigma_1$)
+$\ldots$
+($\vardeclkwd$ $v_n$ $\sigma_n$)
+($\vardeclkwd$ $v'_n$ $\sigma_n$)
+($\constraintkwd$ (=> ($S_{pre}$ $v_1$ $\ldots$ $v_n$) ($S$ $v_1$ $\ldots$ $v_n$)))
+($\constraintkwd$ (=> (and ($S$ $v_1$ $\ldots$ $v_n$) ($S_{trans}$ $v_1$ $\ldots$ $v_n$ $v'_1$ $\ldots$ $v'_n$)) ($S$ $v'_1$ $\ldots$ $v'_n$)))
+($\constraintkwd$ (=> ($S$ $v_1$ $\ldots$ $v_n$) ($S_{post}$ $v_1$ $\ldots$ $v_n$)))
+\end{lstlisting}
 where $v_1, v'_1, \ldots, v_n, v'_n$ are fresh symbols.
 
+\item \sygusinline|($\constraintchckwd$ (($x_1$ $\sigma_1$) $\ldots$ ($x_m$ $\sigma_m$)) $t_1$ $\ldots$ $t_n$ $t_{head}$)|
+
+This command adds a constrained Horn clause (CHC) constraint to the current state.
+Concretely, this command adds $m$ fresh variables $v_1, \ldots, v_m$ to the list of universal variables,
+and a constraint $((\lambda x_1,\ldots,x_m \ldotp t_{body})\ v_1\ \ldots\ v_m) \implies ((\lambda x_1,\ldots,x_m \ldotp t_{head})\ v_1\ \ldots\ v_m)$, where
+$$
+t_{body} \triangleq
+\begin{cases}
+  (t_1 \land \ldots \land t_n)  & n > 0\\
+  \truekwd                      & n = 0
+\end{cases}
+$$
+This is a generalization of the $\constraintinvkwd$ command,
+that allows for specifying problems with multiple, possibly interdependent,
+invariants to be synthesized.
+
+Let $\mathcal{S}$ be the set of symbols to be synthesized.
+We will say that a term $t$ is \emph{atomic} with respect to $\mathcal{S}$ if either
+(a) $t$ does not contain any symbol that belongs to $\mathcal{S}$, or
+(b) $t$ is of the form \sygusinline|($S$ $u_1$ $\ldots$ $u_k$)|
+for some predicate symbol $S \in \mathcal{S}$ and $u_1, \ldots, u_k$ are variables.
+Then, a $\constraintchckwd$ command is well-formed with respect to $\mathcal{S}$ iff:
+\begin{enumerate}
+  \item The \emph{head} of the CHC (i.e., $t_{head}$) is an atomic formula w.r.t. $\mathcal{S}$
+  \item Each conjunct in the \emph{body} of the CHC (i.e., each of $t_1, \ldots, t_n$) is an atomic formula w.r.t. $\mathcal{S}$
```

Comment:
Given my suggestion above, this would say something like "if `t_body` has form `(and t_1 .... t_n)` where ..."